plugins/modules/mso_schema_template_migrate.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Anvitha Jain (@anvitha-jain) <anvjain@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_schema_template_migrate
+short_description: Migrate Bridge Domains (BDs) and EPGs between templates
+description:
+- Migrate BDs and EPGs between templates of same and different schemas.
+author:
+- Anvitha Jain (@anvitha-jain)
+options:
+  schema:
+    description:
+    - The name of the schema.
+    type: str
+    required: yes
+  template:
+    description:
+    - The name of the template.
+    type: str
+    required: yes
+  bds:
+    description:
+    - The name of the BDs to migrate.
+    type: list
+    elements: str
+  epgs:
+    description:
+    - The name of the EPGs to migrate.
+    type: list
+    elements: dict
+  target_schema:
+    description:
+    - The name of the target_schema.
+    type: str
+    required: yes
+  target_template:
+    description:
+    - The name of the target_template.
+    type: str
+    required: yes
+  state:
+    description:
+    - Use C(present) for adding .
+    type: str
+    default: present
+extends_documentation_fragment: cisco.mso.modules
+'''
+
+EXAMPLES = r'''
+- name: Migration of objects between templates of same schema
+  mso_schema_template_migrate:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    target_schema: Schema 1
+    target_template: Template 2
+    bds:
+    - BD
+    epgs:
+    - epg: EPG1
+    anp: ANP
+    state: present
+  delegate_to: localhost
+
+- name: Migration of objects between templates of different schema
+  mso_schema_template_migrate:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    target_schema: Schema 2
+    target_template: Template 2
+    bds:
+    - BD
+    epgs:
+    - epg: EPG1
+    anp: ANP
+    state: present
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec, mso_reference_spec, mso_subnet_spec, mso_dhcp_spec
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type='str', required=True),
+        template=dict(type='str', required=True),
+        bds=dict(type='list', elements='str'),  # This parameter is not required for querying all objects
```

Comment:
do we support query in this module?
if not, you may want to remove comment line here

Reply:
I think it would be good to add query.
Once migration is done, user may want to check that the migration has indeed taken place.

Reply:
Not sure they need a query on this module. They can use the query on EPG, BD or event template/schemas.